engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/EpbContext.java

```diff
@@ -32,20 +36,65 @@ public EpbContext(TruffleLanguage.Env env) {
   }
 
   /**
-   * Initializes the context. No-op in the inner context. Spawns the inner context if called from
-   * the outer context.
+   * Initializes the context.No-op in the inner context. Spawns the inner context if called from the
+   * outer context. Shielded against double initialization.
+   *
+   * @param preInitializeLanguages comma separated list of languages to immediately initialize
    */
-  public void initialize() {
+  public void initialize(String preInitializeLanguages) {
     if (!isInner) {
-      innerContext =
-          new GuardedTruffleContext(
-              env.newInnerContextBuilder()
-                  .initializeCreatorContext(true)
-                  .inheritAllAccess(true)
-                  .config(INNER_OPTION, "yes")
-                  .build(),
-              true);
+      if (innerContext == null) {
+        innerContext =
+            new GuardedTruffleContext(
+                env.newInnerContextBuilder()
+                    .initializeCreatorContext(true)
+                    .inheritAllAccess(true)
+                    .config(INNER_OPTION, "yes")
+                    .build(),
+                true);
+      }
+      initializeLanguages(env, innerContext, preInitializeLanguages);
+    }
+  }
+
+  private static void initializeLanguages(
+      TruffleLanguage.Env environment, GuardedTruffleContext innerContext, String langs) {
+    if (langs == null || langs.isEmpty()) {
+      return;
+    }
+    var log = environment.getLogger(EpbContext.class);
+    log.log(Level.FINE, "Initializing languages {0}", langs);
+    var cdl = new CountDownLatch(1);
+    var run =
+        (Consumer<TruffleContext>)
+            (context) -> {
+              var lock = innerContext.enter(null);
+              log.log(Level.FINEST, "Entering initialization thread");
+              cdl.countDown();
+              try {
+                for (var l : langs.split(",")) {
+                  log.log(Level.FINEST, "Initializing language {0}", l);
+                  long then = System.currentTimeMillis();
+                  var res = context.initializeInternal(null, l);
+                  long took = System.currentTimeMillis() - then;
+                  log.log(
+                      Level.FINE,
+                      "Done initializing language {0} with {1} in {2} ms",
```

Comment:
Isn't this spurious? Logs already have timestamps so one can infer that easily

Reply:
I don't think so. The more information you can provide about some language's initialization, the better. Besides, who wants to calculate something manually if a computer can do it, and is not a performance concern?

Reply:
Please note that the primary usage of this is the `MockLogHandler` in the tests to verify that the initialization really happened. I printed the time just to see how long it takes on my computer. It shouldn't hurt - the `LogRecord` is not converted to string by default.